include/ddc/kernels/splines/spline_evaluator.hpp

```diff
@@ -13,10 +13,23 @@
 
 namespace ddc {
 
+/**
+ * @brief A class to evaluate, derivate or integrate a spline function.
+ *
+ * A class which contains an operator () which can be used to evaluate, derivate or integrate a spline function.
+ *
+ * @tparam ExecSpace The Kokkos execution space on which the spline transform is performed.
+ * @tparam MemorySpace The Kokkos memory space on which the data (interpolation function and splines coefficients) is stored.
+ * @tparam BSplines The discrete dimension representing the B-splines.
+ * @tparam InterpolationMesh The discrete dimension on which interpolation points are defined.
```

Comment:
You want to call it evaluation points ?

```
In [engineering](https://en.wikipedia.org/wiki/Engineering) and [science](https://en.wikipedia.org/wiki/Science), one often has a number of data points, obtained by [sampling](https://en.wikipedia.org/wiki/Sampling_(statistics)) or [experimentation](https://en.wikipedia.org/wiki/Experimentation), which represent the values of a function for a limited number of values of the [independent variable](https://en.wikipedia.org/wiki/Dependent_and_independent_variables). It is often required to interpolate; that is, estimate the value of that function for an intermediate value of the independent variable.
```

Maybe this is because we have some problem to agree for a terminology, maybe we do not have the same definiton for an interpolation.

Reply:
To me `use a spline builder + spline evaluator to approximate a function at a different set of points` is exactly what an interpolation is.

Reply:
I think you are ignoring the case where we have a spline but are not interpolating a known function. For example when we obtain a spline by solving a FEM equation. A spline is itself a function and it is not only useful as an approximation of another function.

In your definition you have:
> one often has a number of data points which represent the values of a function for a limited number of values of the [independent variable](https://en.wikipedia.org/wiki/Dependent_and_independent_variables).

In the case where the spline is not obtained via the SplineBuilder this section does not apply to splines. This is why I don't think the terminology is relevant in this file.

Further even supposing we are using the builder your implementation doesn't force us to evaluate the function at the points that were used to create the spline.


> To me use a spline builder + spline evaluator to approximate a function at a different set of points is exactly what an interpolation is.

I agree but this is not an interpolator it is a spline evaluator and an interpolator is only one way of using the evaluator. I also suspect that it is only an interpolator if these "evaluation_points" match the "interpolation_points". So I think the current definition is accurate in the context of an interpolator but not accurate in any other context

Reply:
This can be closed now that interpolation is renamed evaluation ?